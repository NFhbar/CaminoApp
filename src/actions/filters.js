// SET_TEXT_FILTER
export const setTextFilter = (text = '') => ({
  type: 'SET_TEXT_FILTER',
  text
});

// SORT_BY_SALES
export const sortBySales = () => ({
  type: 'SORT_BY_SALES'
});

// SORT_BY_AMOUNT
export const sortByAmount = () => ({
  type: 'SORT_BY_AMOUNT'
});
