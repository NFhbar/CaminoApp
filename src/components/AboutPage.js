import React from 'react';
import { NavLink } from 'react-router-dom';
import { row, column } from 'reactstrap';

const AboutPage = () => (
  <div className="row card-body">
    <div className="col-sm-12">
      <div className="jumbotron">
        <h1 className="display-4">React-Redux Camino Financial App</h1>
        <p className="lead">This App is a take-home assignment built by
          <a href="https://www.linkedin.com/in/nicolas-frega/"> Nicolas Frega </a>
          for <a href="https://www.caminofinancial.com/"> Camino Financial </a>.
         </p>
        <hr className="my-4"/>
        <p>The App is built with React, Redux, Bootstrap, and Chartjs. It is deployed using Heroku. It mimics some
        of the behavior of a store App in which the admin can add items to sell at a
        given price. Added items are randomly generated a sold amount for displaying purposes. Items can be sorted by price or amount sold.
        The items are then rendered dynamically to the page. The charts are statics
        since no real sales data is present.</p>
        <p className="lead">
          <a className="btn btn-primary btn-lg" href="https://github.com/NFhbar/CaminoApp" role="button">View github</a>
        </p>
      </div>
    </div>
  </div>
);

export default AboutPage;
