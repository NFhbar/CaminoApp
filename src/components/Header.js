import React from 'react';
import { NavLink } from 'react-router-dom';
import { row, column } from 'reactstrap';

const Header = () => (
  <div className="row card-header">
    <div className="col-sm-12">
        <h1 className="card-title" id="header-title">Camino Financial React App</h1>
        <div className="row">
          <div className="col-md-4">
            <NavLink to="/" className="btn btn-primary" activeClassName="is-active" exact={true}>Dashboard</NavLink>
          </div>
          <div className="col-md-4">
            <NavLink to="/create" className="btn btn-primary" activeClassName="is-active">Add Item</NavLink>
          </div>
          <div className="col-md-4">
            <NavLink to="/about" className="btn btn-primary" activeClassName="is-active">About</NavLink>
          </div>
        </div>
    </div>
  </div>
);

export default Header;
