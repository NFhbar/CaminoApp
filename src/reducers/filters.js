// Filters Reducer
const filtersReducerDefaultState = {
  text: '',
  sortBy: 'amount'
}
//state = filtersReducerDefaultState
export default (state = filtersReducerDefaultState, action) => {
  switch (action.type) {
    case 'SET_TEXT_FILTER':
      return {
        ...state,
        text: action.text
      };
    case 'SORT_BY_AMOUNT':
      return {
        ...state,
        sortBy: 'amount'
      };
    case 'SORT_BY_SALES':
      return {
        ...state,
        sortBy: 'sales'
      };
    default:
      return state;
  }
};
