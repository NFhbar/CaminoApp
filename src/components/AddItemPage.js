import React from 'react';
import { connect } from 'react-redux';
import { row, column } from 'reactstrap';
import ItemForm from './ItemForm'
import { addItem } from '../actions/items';

export class AddItemPage extends React.Component {
  onSubmit = (item) => {
    this.props.addItem(item);
    this.props.history.push('/');
  };
  render() {
    return (
      <div className="row card-body">
        <div className="col-sm-12">
          <h3>Add an item to your store!</h3>
          <ItemForm
            onSubmit={this.onSubmit}
          />
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  addItem: (item) => dispatch(addItem(item))
});

export default connect(undefined, mapDispatchToProps)(AddItemPage);
