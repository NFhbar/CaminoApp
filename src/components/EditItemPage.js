import React from 'react';
import { connect } from 'react-redux';
import { row, column } from 'reactstrap';
import ItemForm from './ItemForm';
import { editItem, startRemoveItem } from '../actions/items';

export class EditItemPage extends React.Component {
  onSubmit = (item) => {
    this.props.editItem(this.props.item.id, item);
    this.props.history.push('/');
  };
  onRemove = () => {
    this.props.startRemoveItem({ id: this.props.item.id });
    this.props.history.push('/');
  };
  render() {
    return (
      <div className="row" id="edit-item-page">
        <div className="col-sm-12">
          <ItemForm
            item={this.props.item}
            onSubmit={this.onSubmit}
          />
          <button type="submit" id="btn-remove" className="btn btn-md btn-danger" onClick={this.onRemove}>Remove Item</button>
        </div>
      </div>
    );
  }
};

const mapStateToProps = (state, props) => ({
  item: state.items.find((item) => item.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
  editItem: (id, item) => dispatch(editItem(id, item)),
  startRemoveItem: (data) => dispatch(startRemoveItem(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditItemPage);
