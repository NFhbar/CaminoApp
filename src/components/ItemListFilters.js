import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByAmount, sortBySales } from '../actions/filters';

export class ItemListFilters extends React.Component {
  state = {
    calendarFocused: null
  };
  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
  };
  onSortChange = (e) => {
    if (e.target.value === 'amount'){
      this.props.sortByAmount();
    } else if (e.target.value === 'sales'){
      this.props.sortBySales();
    }
  };
  render() {
    return (
      <form>
        <div className="form-group">
          <input
            type="text"
            value={this.props.filters.text}
            onChange={this.onTextChange}
            className="form-control"
            placeholder="seacrh"
          />
          <select
            value={this.props.filters.sortBy}
            onChange={this.onSortChange}
            className="form-control"
          >
            <option value="amount">Price</option>
            <option value="sales">Sales</option>
          </select>
        </div>
      </form>
      );
  }
};

const mapStateToProps = (state) => ({
    filters: state.filters
});

const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  sortByAmount: () => dispatch(sortByAmount()),
  sortBySales: () => dispatch(sortBySales())
});

export default connect(mapStateToProps, mapDispatchToProps)(ItemListFilters);
