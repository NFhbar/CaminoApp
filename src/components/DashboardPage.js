import React from 'react';
import { row, column } from 'reactstrap';
import ItemList from './ItemList';
import ItemPlot from './ItemPlot';
import ItemListFilters from './ItemListFilters';

const DashboardPage = () => (
  <div className="row card-body">
    <div className="col-sm-12">
      <ItemListFilters />
      <ul className="list-group">
        <li className="list-group-item disabled">
          <div className="row">
            <div className="col-sm-4">
              Item
            </div>
            <div className="col-sm-4">
              Price
            </div>
            <div className="col-sm-4">
              Items sold
            </div>
          </div>
        </li>
        <ItemList />
      </ul>
      <ItemPlot />
    </div>
  </div>
);

export default DashboardPage;
