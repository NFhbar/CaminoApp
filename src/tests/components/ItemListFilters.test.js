import React from 'react';
import { shallow } from 'enzyme';
import { ItemListFilters } from '../../components/ItemListFilters';
import { filters, altFilters } from '../fixtures/filters';

let setTextFilter, sortByAmount, wrapper;

beforeEach(() => {
  setTextFilter = jest.fn();
  sortByAmount = jest.fn();
  wrapper = shallow(
    <ItemListFilters
      filters={filters}
      setTextFilter={setTextFilter}
      sortByAmount={sortByAmount}
    />
  );
});

test('should render ItemListFilters correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should render ItemListFilters with alt data correctly', () => {
  wrapper.setProps({
    filters: altFilters
  });
  expect(wrapper).toMatchSnapshot();
});

test('should handle text change', () => {
  const value = 'pants';
  wrapper.find('input').simulate('change', {
    target: { value }
  });
  expect(setTextFilter).toHaveBeenLastCalledWith(value);
});

test('should sort by amount', () => {
  const value = 'amount';
  wrapper.setProps({
    filters: altFilters
  });
  wrapper.find('select').simulate('change', {
    target: { value }
  });
  expect(sortByAmount).toHaveBeenCalled();
});
