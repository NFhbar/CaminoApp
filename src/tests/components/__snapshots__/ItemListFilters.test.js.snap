// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ItemListFilters correctly 1`] = `
<form>
  <div
    className="form-group"
  >
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="seacrh"
      type="text"
      value=""
    />
    <select
      className="form-control"
      onChange={[Function]}
      value="amount"
    >
      <option
        value="amount"
      >
        Price
      </option>
      <option
        value="sales"
      >
        Sales
      </option>
    </select>
  </div>
</form>
`;

exports[`should render ItemListFilters with alt data correctly 1`] = `
<form>
  <div
    className="form-group"
  >
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="seacrh"
      type="text"
      value="pants"
    />
    <select
      className="form-control"
      onChange={[Function]}
      value="amount"
    >
      <option
        value="amount"
      >
        Price
      </option>
      <option
        value="sales"
      >
        Sales
      </option>
    </select>
  </div>
</form>
`;
